src/cloudinaryUploader.js

```diff
@@ -74,10 +74,40 @@ const DEFAULT_REQUEST_OPTIONS = {
   mode: 'cors',
 }
 
-// The default public id creator returns the sanitized name of the file via html escaping.
-// Source: https://support.cloudinary.com/hc/en-us/articles/115001317409--Legal-naming-conventions
+export const FORBIDDEN_PATTERN = /[?&#\\%<>]/gi
+
+/**
+ * The default public id creator returns the sanitized name of the file by
+ * replacing invalid characters (incl. those that may be html escaped).
+ * Cloudinary does _not_ handle html escaping well via the API our their
+ * dashboard so it's best to avoid it entirely.
+ * 
+ * Source: https://support.cloudinary.com/hc/en-us/articles/115001317409--Legal-naming-conventions
+ */
 function defaultCreatePublicId (file) {
-  return encodeURIComponent(file.name)
+  const decodedFileName = decodeFileName(file.name)
+  return sanitizeFileName(decodedFileName)
+}
+
+// Attempts to decode html escaped characters. If this fails, strips characters
+// causing a malformed URI and tries again. If all else fails, returns the original name
+function decodeFileName (name='') {
+  try {
+    try {
+      return decodeURIComponent(name)
+    } catch (e) {
+      return decodeURIComponent(name.replace(/%(?=\D+)/gi, ''))
```

Comment:
Can you give an example of when this fails? Is it a specific case or just playing it safe?

Reply:
It's when you have `%` that doesn't act as an escaped encoding. For example, `100%`. I've included that implicitly as a test case in one of my tests

Reply:
Do you want me to break this out into a separate test? https://github.com/LaunchPadLab/lp-hoc/blob/4c8ebf9eb1e1ea6a516c583a1409e2d5a2e83be2/test/cloudinary-uploader.test.js#L126